homeassistant/components/samsungtv/config_flow.py

```diff
@@ -0,0 +1,177 @@
+"""Config flow for Samsung TV."""
+import socket
+
+from samsungctl import Remote
+from samsungctl.exceptions import AccessDenied, UnhandledResponse
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_ID, CONF_IP_ADDRESS, CONF_NAME
+from homeassistant.components.ssdp import (
+    ATTR_HOST,
+    ATTR_NAME,
+    ATTR_MODEL_NAME,
+    ATTR_MANUFACTURER,
+    ATTR_UDN,
+)
+
+from .const import CONF_MANUFACTURER, CONF_MODEL, DOMAIN, LOGGER, METHODS
+
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str, vol.Required(CONF_NAME): str})
+
+RESULT_AUTH_MISSING = "auth_missing"
+RESULT_SUCCESS = "success"
+RESULT_NOT_FOUND = "not_found"
+RESULT_NOT_SUPPORTED = "not_supported"
+
+
+def _get_ip(host):
+    if host is None:
+        return
+    return socket.gethostbyname(host)
+
+
+async def _async_try_connect(host, name):
+    """Try to connect and check auth."""
+    for method in METHODS:
+        config = {
+            "name": "HomeAssistant",
+            "description": name,
+            "id": "ha.component.samsung",
+            "host": host,
+            "method": method,
+            "port": None,
+            "timeout": None,
+        }
+        try:
+            LOGGER.debug("Try config: %s", config)
+            with Remote(config.copy()):
+                LOGGER.debug("Working config: %s", config)
+                return RESULT_SUCCESS
+        except AccessDenied:
+            LOGGER.debug("Working but denied config: %s", config)
+            return RESULT_AUTH_MISSING
+        except UnhandledResponse:
+            LOGGER.debug("Working but unsupported config: %s", config)
+            return RESULT_NOT_SUPPORTED
+        except (OSError):
+            LOGGER.debug("Failing config: %s", config)
+
+    LOGGER.debug("No working config found")
+    return RESULT_NOT_FOUND
+
+
+class SamsungTVConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Samsung TV config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+
+    def __init__(self):
+        """Initialize flow."""
+        self._host = None
+        self._ip = None
+        self._manufacturer = None
+        self._model = None
+        self._uuid = None
+        self._name = None
+        self._title = None
+
+    def _is_already_configured(self):
+        if any(
+            self._ip == entry.data.get(CONF_IP_ADDRESS)
```

Comment:
Maybe accept an ip address as parameter instead of using the instance attribute? That would make it more flexible.